components/AllocatedWorkers/AllocatedWorkers.tsx

```diff
@@ -7,14 +7,34 @@ import Button from 'components/Button/Button';
 import { useAuth } from 'components/UserContext/UserContext';
 import { useAllocatedWorkers } from 'utils/api/allocatedWorkers';
 
-import { User } from 'types';
+import { Resident, User } from 'types';
 
 interface Props {
-  id: number;
+  person: Resident;
 }
 
-const AllocatedWorkers = ({ id }: Props): React.ReactElement => {
-  const { data: { allocations } = {}, error } = useAllocatedWorkers(id);
+const canUserAllocateWorkerToPerson = (user: User, person: Resident) => {
+  if (user.hasAdminPermissions || user.hasDevPermissions) {
+    return true;
+  }
+
+  if (user.hasChildrenPermissions && person.contextFlag === 'C') {
```

Comment:
Should we also check `(user.hasAllocationsPermissions || false)` here too?

Reply:
The `hasAllocationPermissions` only applies to adults, for childrens anyone can allocate irrespective of this role. So nice and confusing, haha 😱

Reply:
I added a comment to clarify this @Renny4Real

Reply:
🙌🏾